components/org.wso2.carbon.identity.application.authenticator.hypr.rest/org.wso2.carbon.identity.application.authenticator.hypr.rest.v1/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/rest/v1/core/ServerHYPRAuthenticatorService.java

```diff
@@ -0,0 +1,269 @@
+
+        if (!(authenticatorProperties.containsKey(HYPRConstants.BASE_URL) &&
+                authenticatorProperties.containsKey(HYPRConstants.API_TOKEN))) {
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    HYPRConstants.ErrorMessage.SERVER_ERROR_INVALID_AUTHENTICATOR_CONFIGURATIONS);
+        }
+
+        Map<String, String> hyprConfigurations = new HashMap<>();
+        hyprConfigurations.put(HYPRConstants.BASE_URL, authenticatorProperties.get(HYPRConstants.BASE_URL));
+        hyprConfigurations.put(HYPRConstants.API_TOKEN, authenticatorProperties.get(HYPRConstants.API_TOKEN));
+
+        return hyprConfigurations;
+    }
+
+    /**
+     * Extract the user authentication properties such as authentication status and request ID from the context.
+     *
+     * @param sessionContext The authentication context for the given session key.
+     */
+    private Map<String, String> getHyprAuthenticationProperties(AuthenticationContext sessionContext) {
+
+        if (sessionContext.getProperty(HYPRConstants.AUTH_STATUS) == null ||
+                sessionContext.getProperty(HYPRConstants.AUTH_REQUEST_ID) == null) {
+            throw handleError(Response.Status.INTERNAL_SERVER_ERROR,
+                    HYPRConstants.ErrorMessage.SERVER_ERROR_INVALID_AUTHENTICATION_PROPERTIES);
+        }
+
+        Map<String, String> hyprAuthenticationProperties = new HashMap<>();
+        hyprAuthenticationProperties.put(HYPRConstants.AUTH_STATUS,
+                (String) sessionContext.getProperty(HYPRConstants.AUTH_STATUS));
+        hyprAuthenticationProperties.put(HYPRConstants.AUTH_REQUEST_ID,
+                (String) sessionContext.getProperty(HYPRConstants.AUTH_REQUEST_ID));
+        return hyprAuthenticationProperties;
+    }
+
+    /**
+     * Convert the HTTPResponse to a json node.
+     *
+     * @param response A HTTPResponse object received from API call.
+     * @throws IOException
+     */
+    private JsonNode toJsonNode(CloseableHttpResponse response) throws IOException {
+
+        JsonNode rootNode = null;
+        HttpEntity entity = response.getEntity();
+        if (entity != null) {
+            // Convert to a string
+            String result = EntityUtils.toString(entity);
+            rootNode = new ObjectMapper().readTree(new StringReader(result));
+        }
+        return rootNode;
+    }
+
+    /**
+     * Handle invalid input.
+     *
+     * @param errorEnum
+     * @param data
+     * @return APIError
+     */
+    public APIError handleInvalidInput(HYPRConstants.ErrorMessage errorEnum, String... data) {
+
+        return handleError(Response.Status.BAD_REQUEST, errorEnum);
+    }
+
+    /**
+     * Handle Exceptions.
+     *
+     * @param e
+     * @param errorEnum
+     * @return
+     */
+    private APIError handleException(Exception e, HYPRConstants.ErrorMessage errorEnum, boolean isClientException,
```

Comment:
Generally, it is not required to add method comments for private methods. But if you need to add a comment, let's add param and return descriptions

Reply:
Fixed with [38a90ea](https://github.com/wso2-extensions/identity-outbound-auth-hypr/pull/1/commits/38a90eae1349a0988459ee5205d4d728aec92bca)